src/state/ObservableScope.test.ts

```diff
@@ -102,3 +102,52 @@ describe("Epoch", () => {
     s$.complete();
   });
 });
+
+describe("Reconcile", () => {
+  it("should wait for cleanup to complete before processing next Value", async () => {
+    vi.useFakeTimers();
+
+    const scope = new ObservableScope();
+    const cleanUp1 = Promise.withResolvers<void>();
+    const cleanUp2 = vi.fn();
+
+    const behavior$ = new BehaviorSubject<number>(1);
+    let lastProcessed = 0;
+
+    scope.reconcile<number>(
+      behavior$,
+      async (value: number): Promise<(() => Promise<void>) | void> => {
+        lastProcessed = value;
+        if (value === 1) {
+          return Promise.resolve(async (): Promise<void> => cleanUp1.promise);
+        } else if (value === 2) {
+          return Promise.resolve(async (): Promise<void> => {
+            cleanUp2();
+            return Promise.resolve(undefined);
+          });
+        }
+        return Promise.resolve();
+      },
+    );
```

Comment:
```suggestion
    const setup = vi.fn().mockImplementation(async ()=>await sleep(100)));
    const cleanup = vi.fn().mockImplementation(async ()=>await sleep(100)))
    scope.reconcile(
      behavior$,
      async () => {
       await setup();
       return cleanup
      },
    );
```
Could be enough for the `reconcile` call. Then the test can be fully controlled by: `behavior$.next` and `vi.advanceTimersByTimeAsync(100);` as outlined below.